src/depackers/stb_vorbis.c

```diff
@@ -5230,50 +5517,45 @@ int stb_vorbis_decode_filename(char *filename, int *channels, short **output)
 }
 #endif // NO_STDIO
 
-int stb_vorbis_decode_memory(uint8 *mem, int len, int *channels, short **output)
+int stb_vorbis_decode_memory(const uint8 *mem, int len, int *channels, int *sample_rate, short **output)
 {
    int data_len, offset, total, limit, error;
    short *data;
-   stb_vorbis *v;
-
-   D_(D_INFO "vorbis_decode_memory (len=%d)", len);
-
-   v = stb_vorbis_open_memory(mem, len, &error, NULL);
+   stb_vorbis *v = stb_vorbis_open_memory(mem, len, &error, NULL);
    if (v == NULL) return -1;
    limit = v->channels * 4096;
    *channels = v->channels;
+   if (sample_rate)
+      *sample_rate = v->sample_rate;
    offset = data_len = 0;
    total = limit;
-   D_(D_INFO "total=%d\n", total);
    data = (short *) malloc(total * sizeof(*data));
    if (data == NULL) goto error;
    for (;;) {
       int n = stb_vorbis_get_frame_short_interleaved(v, v->channels, data+offset, total-offset);
-      if (n < 0)  goto error;
+      if (n < 0) goto error;
       if (n == 0) break;
       data_len += n;
       offset += n * v->channels;
-      D_(D_INFO "offset=%d, limit=%d, total=%d", offset, limit, total);
       if (offset + limit > total) {
-	 short *data2;
-	 total *= 2;
-	 data2 = (short *) realloc(data, total * sizeof(*data));
-	 if (data2 == NULL) goto error;
-	 data = data2;
+         short *data2;
+         total *= 2;
+         data2 = (short *) realloc(data, total * sizeof(*data));
+         if (data2 == NULL) goto error;
+         data = data2;
       }
    }
-   stb_vorbis_close(v);
    *output = data;
+   stb_vorbis_close(v);
    return data_len;
 
 error:
    free(data);
    stb_vorbis_close(v);
    return -2;
 }
-#endif
+#endif // STB_VORBIS_NO_INTEGER_CONVERSION
 
-#if 0
 int stb_vorbis_get_samples_float_interleaved(stb_vorbis *f, int channels, float *buffer, int num_floats)
```

Comment:
`stb_vorbis_get_samples_float_interleaved` and `stb_vorbis_get_samples_float`
aren't used by us, hence they were commented out. Maybe should do that again?

Reply:
Seconded—we don't use these.